src/server/es/aggs.js

```diff
@@ -196,6 +219,9 @@ export const numericGlobalStats = async (
  * @param {object} param1.filterSelf - only valid if to avoid filtering the same aggregation field
  * @param {object} param1.defaultAuthFilter - once param1.filter is empty,
  *                                            use this auth related filter instead
+ * @param {object} param1.nestedAggFields - fields for sub-aggregations
+ *                                          (terms and/or missing aggregation)
+ * @param {object} param1.nestedPath - path info used by nested aggregation
  */
 export const numericHistogramWithFixedRangeStep = async (
```

Comment:
Tangential to your code, but I think there might be a bug in this function if you call it with rangeStep=0 -- it seems like due to the while loop on this line it will loop infinitely: https://github.com/uc-cdis/guppy/pull/83/files#diff-2375bd5a21772ae861e3c59ea9cf6fb8L249

Reply:
Also one more, can you pass both nestedPath and nestedAggFields to this function? (Can you do both nested aggregation and sub-aggregations at the same time?)

Reply:
> Tangential to your code, but I think there might be a bug in this function if you call it with rangeStep=0 -- it seems like due to the while loop on this line it will loop infinitely: https://github.com/uc-cdis/guppy/pull/83/files#diff-2375bd5a21772ae861e3c59ea9cf6fb8L249

there is a guard here https://github.com/uc-cdis/guppy/blob/feat/nested-agg/src/server/es/aggs.js#L439 stopping you from doing that 😃

Reply:
> Also one more, can you pass both nestedPath and nestedAggFields to this function? (Can you do both nested aggregation and sub-aggregations at the same time?)

Yes you could! Here is an example
Query:
```
{
	"query": "query ($filter: JSON, $nestedAggFields: JSON) {\n    _aggregation {\n      subject (filter: $filter, nestedAggFields: $nestedAggFields, accessibility: all) {\n        visits {\n         follow_ups {\n        follow_up_label {\n            histogram {\n              key\n              count\n  termsFields {\n  field\n    terms {\n  key\n  count\n    }\n   }\n          }\n          }\n   }\n     }\n      }\n    }\n  }",
	"variables": {
		"filter": {
			"AND": [
				{
					"nested": {
						"path": "visits",
						"AND": [
							{
								">=": {
									"days_to_visit": 1
								}
							}
						]
					}
				}
			]
		},
		"nestedAggFields": {
			"termsFields": [
				"visits.follow_ups.days_to_follow_up"
			]
		}
	}
}
```

Result:
```
{
  "data": {
    "_aggregation": {
      "subject": {
        "visits": {
          "follow_ups": {
            "follow_up_label": {
              "histogram": [
                {
                  "key": "flup_lbl_3",
                  "count": 29,
                  "termsFields": [
                    {
                      "field": "visits.follow_ups.days_to_follow_up",
                      "terms": [
                        {
                          "key": "3",
                          "count": 29
                        }
                      ]
                    }
                  ]
                },
                {
                  "key": "flup_lbl_1",
                  "count": 21,
                  "termsFields": [
                    {
                      "field": "visits.follow_ups.days_to_follow_up",
                      "terms": [
                        {
                          "key": "1",
                          "count": 21
                        }
                      ]
                    }
                  ]
                },
                {
                  "key": "flup_lbl_2",
                  "count": 19,
                  "termsFields": [
                    {
                      "field": "visits.follow_ups.days_to_follow_up",
                      "terms": [
                        {
                          "key": "2",
                          "count": 19
                        }
                      ]
                    }
                  ]
                }
              ]
            }
          }
        }
      }
    }
  }
}
```

Reply:
oops sorry about the ugly format 🤦 should have preview it before posting